lib/config/configitem.cpp

```diff
@@ -470,7 +470,12 @@ bool ConfigItem::CommitNewItems(const ActivationContext::Ptr& context, WorkQueue
 					return;
 
 				ip.first->Commit(ip.second);
-				committed_items++;
+
+				{
+					// Prevent committed_items from being accessed by multiple threads asynchronously
+					std::unique_lock<std::mutex> lock(m_Mutex);
+					committed_items++;
+				}
```

Comment:
Good catch, but I prefer an atomic variable unless our colleague knows better...

Reply:
Yeah, go for `std::atomic`